lib/modem_info/modem_info.c

```diff
@@ -159,32 +197,49 @@ static const struct modem_info_data iccid_data = {
 	.data_type = AT_PARAM_TYPE_STRING,
 };
 
+static const struct modem_info_data systemmode_data = {
+	.cmd = AT_CMD_SYSTEMMODE,
+	.param_index = SYSTEMMODE_PARAM_INDEX,
+	.param_count = SYSTEMMODE_PARAM_COUNT,
+	.data_type = AT_PARAM_TYPE_NUM_SHORT,
+};
+
 static const struct modem_info_data *const modem_data[] = {
-	[MODEM_INFO_RSRP] = &rsrp_data,
-	[MODEM_INFO_BAND] = &band_data,
-	[MODEM_INFO_MODE] = &mode_data,
-	[MODEM_INFO_OPERATOR] = &operator_data,
-	[MODEM_INFO_CELLID] = &cellid_data,
-	[MODEM_INFO_IP_ADDRESS] = &ip_data,
-	[MODEM_INFO_UICC] = &uicc_data,
-	[MODEM_INFO_BATTERY] = &battery_data,
-	[MODEM_INFO_TEMP] = &temp_data,
-	[MODEM_INFO_FW_VERSION] = &fw_data,
-	[MODEM_INFO_ICCID] = &iccid_data,
+	[MODEM_INFO_RSRP]	= &rsrp_data,
+	[MODEM_INFO_CUR_BAND]	= &band_data,
+	[MODEM_INFO_SUP_BAND]	= &band_sup_data,
```

Comment:
The names can be aligned bewteen macros and structs. `SUP_BAND` vs `band_sup`

Reply:
Fixed.